code/modules/vore/smoleworld/smoleworld_vr.dm

```diff
@@ -0,0 +1,407 @@
+	drop_sound = 'sound/items/drop/device.ogg'
+	pickup_sound = 'sound/items/pickup/device.ogg'
+	starts_with = list( /obj/item/stack/material/smolebricks,
+	/obj/item/stack/material/smolebricks, /obj/item/stack/material/smolebricks, /obj/item/stack/material/smolebricks, /obj/item/stack/material/smolebricks,
+	/obj/item/stack/material/smolebricks, /obj/item/stack/material/smolebricks, /obj/item/stack/material/smolebricks, /obj/item/stack/material/smolebricks,
+	/obj/item/stack/material/smolebricks, /obj/item/stack/material/smolebricks, /obj/item/stack/material/smolebricks, /obj/item/stack/material/smolebricks
+	)
+
+//Track code
+//defineing actions
+/obj/structure/smoletrack
+	icon = 'icons/vore/smoleworld_vr.dmi'
+	density = 0
+
+/obj/structure/smoletrack/attack_hand(mob/user)
+	if(user.a_intent == I_DISARM)
+		if(ismouse(usr) || (isobserver(usr) && !config.ghost_interaction))
+			return
+		to_chat(user, "<span class='notice'>[src] was dismantaled into bricks.</span>")
+		playsound(src, 'sound/items/smolesmallbuild.ogg', 50, 1, -1, volume_channel = VOLUME_CHANNEL_MASTER)
+		var/turf/simulated/floor/F = get_turf(src)
+		if(istype(F))
+			new /obj/item/stack/material/smolebricks(F)
+		qdel(src)
+
+/obj/structure/smoletrack/verb/rotate_clockwise()
+	set name = "Rotate Road Clockwise"
+	set category = "Object"
+	set src in oview(1)
+	if(ismouse(usr) || (isobserver(usr) && !config.ghost_interaction))
+		return
+	src.set_dir(turn(src.dir, 270))
+
+// Road pieces
+
+/obj/structure/smoletrack/roadS
+	name = "road straight piece"
+	icon = 'icons/vore/smoleworld_vr.dmi'
+	icon_state = "carstraight"
+	desc = "A long set of tiny road."
+	anchored = 1
+
+/obj/structure/smoletrack/roadT
+	name = "road threeway piece"
+	icon = 'icons/vore/smoleworld_vr.dmi'
+	icon_state = "carthreeway"
+	desc = "A tiny threeway road piece."
+	anchored = 1
+
+/obj/structure/smoletrack/roadturn
+	name = "road turn piece"
+	icon = 'icons/vore/smoleworld_vr.dmi'
+	icon_state = "carturn"
+	desc = "A tiny turn road piece."
+	anchored = 1
+
+/obj/structure/smoletrack/roadF
+	name = "road fourway piece"
+	icon = 'icons/vore/smoleworld_vr.dmi'
+	icon_state = "carfourway"
+	desc = "A fourway road piece."
+	anchored = 1
+
+//buildings code
+//Defining building actions
+/obj/structure/smolebuilding
+	icon = 'icons/vore/smoleworld_vr.dmi'
+	density = 1
+	anchored = 1
+	var/health = 75
+	var/damage
+//makes it so buildings can be dismaintaled or GodZilla style attacked
+/obj/structure/smolebuilding/attack_hand(mob/user)
+	if(user.a_intent == I_DISARM)
+		if(ismouse(usr) || (isobserver(usr) && !config.ghost_interaction))
+			return
+		to_chat(user, "<span class='notice'>[src] was dismantaled into bricks.</span>")
+		playsound(src, 'sound/items/smolesmallbuild.ogg', 50, 1, -1, volume_channel = VOLUME_CHANNEL_MASTER)
+		var/turf/simulated/floor/F = get_turf(src)
+		if(istype(F))
+			new /obj/item/stack/material/smolebricks(F)
+			new /obj/item/stack/material/smolebricks(F)

```

Comment:
```suggestion
		if(!isnull(loc))
			new /obj/item/stack/material/smolebricks(loc)
			new /obj/item/stack/material/smolebricks(loc)
```
So you don't lose bricks if it's not on a simulated floor.

Reply:
ah... I knew that was a thing... at least for holo items. didn't realize that could work in reverse,